File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.cpp
// Last updated: 8/4/2026, 7:17:31 PM
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n = nums.size();
        map<int , int>mpp;

        for(int i = 0 ; i < n ; i++){
            if(mpp.find(nums[i]) != mpp.end()){
                return nums[i];
            }
            mpp[nums[i]]++;
        }
        return -1;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.java
// Last updated: 8/4/2026, 7:17:37 PM
class Solution {
    public int findDuplicate(int[] nums) {
       int n = nums.length;
        int arr[] = new int [n+1];

        for(int i = 0 ; i < n ; i++){
            arr[nums[i]]++;
        }
        for(int i = 1 ; i <= n ; i++){
            if(arr[i] >= 2) return i;
        }
        return -1;
    }
}